### PR TITLE
AO3-6284 Correct message when adding bookmark to collection using Add To Collection option

### PR DESCRIPTION
**Pull request by saltlas**
# Pull Request Checklist

* [x] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [x] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [x] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [x] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [x] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6284

## Purpose

What does this PR do?
Displays "bookmark" when appropriate instead of "work" as per Jira issue.
edit 28/5/23: it also moves all strings from the controllers into en.yml, and changes some cucumber features to account for different pluralization structures that result.
## Testing Instructions

How can the Archive's QA team verify that this is working as you intended?

As per Jira, have also added a cucumber test that catches the bug.

## References

Are there other relevant issues/pull requests/mailing list discussions?

## Credit

What name and pronouns should we use to credit you in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)? salt, they/them

### CODE REVIEW

File: app/controllers/collection_items_controller.rb [hunk 1]
```diff
@@ -144,8 +144,12 @@ def create
       end
     end
     unless unapproved_collections.empty?
+      collection_item_type = "work"
+      if params[:bookmark_id]
+        collection_item_type = "bookmark"
+      end
       flash[:notice] ||= ""
-      flash[:notice] += ts(" You have submitted your work to #{unapproved_collections.size > 1 ? "moderated collections (%{all_collections}). It will not become a part of those collections" : "the moderated collection '%{all_collections}'. It will not become a part of the collection"} until it has been approved by a moderator.", all_collections: unapproved_collections.map { |f| f.title }.join(', '))
+      flash[:notice] += ts(" You have submitted your #{collection_item_type} to #{unapproved_collections.size > 1 ? "moderated collections (%{all_collections}). It will not become a part of those collections" : "the moderated collection '%{all_collections}'. It will not become a part of the collection"} until it has been approved by a moderator.", all_collections: unapproved_collections.map { |f| f.title }.join(', '))
```
[the remark this answers is not in the document]
Comment (designateddolt) [reply app/controllers/collection_items_controller.rb:152]: Could someone please explain this and the other hound message, I don't understand, sorry. Also please let me know if the Cucumber test step is written poorly or is unnecessary altogether and I'll be happy to remove it.

[the remark this answers is not in the document]
Comment (potpotkettle) [reply app/controllers/collection_items_controller.rb:152]: For the first message, does [this](https://docs.rubocop.org/rubocop/cops_style.html#stylestringliteralsininterpolation) help?

I think those messages originally come from Rubocop. You can search for each code starting with "Style/" at https://docs.rubocop.org/rubocop/cops_style.html .

**Comment by designateddolt** [comment]
Ok, I've fixed the style issues on this page and also switched it from ts() to t() internationalization. Might be of interest to people that some of the modified parts of the code in bookmarks_controller and collection_items_controller are basically identical (except that bookmarks_controller is missing a message about being invited to collections, which strikes me as a little odd if all of the other code is identical since i'm under the assumption they do the same thing, but I don't know for sure so I'll leave it for someone with more knowledge to determine) in case it's possible/desirable to merge them into one instance of the code.

**Comment by potpotkettle** [comment]
Some of the errors are from rspec, not cucumber.

https://github.com/otwcode/otwarchive/runs/7666670914?check_suite_focus=true#step:14:3449

```
       expected: "You don't have permission to do that, sorry!"
            got: "You don't have permission to see that, sorry!"
```

It looks like we need to revert "see" to "do" in a message, or modify the test if the message is supposed to change.

(I'm just someone stopping by, and I believe others will do more in-depth reviews later. I hope this comment still helps in the meantime.)

**Comment by designateddolt** [comment]
> Some of the errors are from rspec, not cucumber.
> 
> https://github.com/otwcode/otwarchive/runs/7666670914?check_suite_focus=true#step:14:3449
> 
> ```
>        expected: "You don't have permission to do that, sorry!"
>             got: "You don't have permission to see that, sorry!"
> ```
> 
> It looks like we need to revert "see" to "do" in a message, or modify the test if the message is supposed to change.
> 
> (I'm just someone stopping by, and I believe others will do more in-depth reviews later. I hope this comment still helps in the meantime.)

Thank you!

**Comment by designateddolt** [comment]
Oops - sorry, didn't mean to request a review from you since you said you were just stopping by - was trying to request a review from someone else since i suspect this hasn't been looked at by anyone else since it opened since you're listed as the assigned reviewer.

**Comment by sarken** [comment]
I've removed the pending review request for you! (I have no idea if it's something you can do -- it might need to be done by maintainers -- but I did so by unchecking potpotkettle's name in the gear menu by "Reviewers.") 

Unfortunately, the real reason no one has stopped by to review is just that we're a bit shorthanded at the moment. Thanks for bearing with us!

**Comment by redsummernight** [comment]
There are merge conflicts here now, unfortunately. Can you update the PR? (No rush, though!)

Can you also remove `opendoors_result.txt` from the PR? (I'll add that to .gitignore in a separate change.)

**Comment by saltlas** [comment]
so i've just had a look at this again and i was definitely a bit newer than i am now when i wrote this. the pull is quite out of scope for what the jira issue is asking for - i'm pretty sure the reason i decided to shunt everything in the controllers into the yml file is because i'd been told to do something similar on a different pull that was using ts() and thought it'd be helpful, but i understand if that's not what's desired one an issue that's asking for like, two lines changed. since there are merge conflicts again that would require me to fiddle with the yml and controller and features to resolve, i figured i'd ask - is it preferred if i resolve them and keep the changes i've made here, or close the pull and start again and just change those two words and leave everything else alone?

**Comment by saltlas** [comment]
have been having some trouble working on PRs lately since ive been having computer trouble and uni things going on. i still plan to redo this PR but it might not be for a little bit longer, at least till I get a new laptop and catch up on stuff.

**Comment by sarken** [comment]
Thanks for the update!

**Pull request closed without merging**